server/src/main/java/org/opentosca/toscana/plugins/kubernetes/docker/image/DockerImageBuilder.java

```diff
@@ -0,0 +1,103 @@
+package org.opentosca.toscana.plugins.kubernetes.docker.image;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.opentosca.toscana.core.plugin.PluginFileAccess;
+
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerCertificateException;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.DockerException;
+import com.spotify.docker.client.ProgressHandler;
+import com.spotify.docker.client.messages.ProgressMessage;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ This class allows the automatic building of a dockerfile
+ <p>
+ if a docker daemon is available.
+ */
+public class DockerImageBuilder implements ProgressHandler {
+
+    private final Logger logger;
+    private final String tag;
+    private final String dockerWorkDir;
+    private final PluginFileAccess access;
+
+    /**
+     @param tag           the tag ("name") that the resulting image should have
+     @param dockerWorkDir The working directory (relative to transformation root) that contains the dockerfile
+     @param access        the PluginFileAccess to read and write data
+     @param logger        the logger to log the Docker build responses to (their level is info)
+     */
+    public DockerImageBuilder(String tag, String dockerWorkDir, PluginFileAccess access, Logger logger) {
+        this.logger = logger;
+        this.tag = tag;
+        this.dockerWorkDir = dockerWorkDir;
+        this.access = access;
+    }
+
+    /**
+     Just like the other constructor
+     but the logger gets built using <code>LoggerFactory.getLogger(DockerImageBuilder.class)</code>
+     */
+    public DockerImageBuilder(String tag, String dockerWorkDir, PluginFileAccess access) {
+        this(
+            tag,
+            dockerWorkDir,
+            access,
+            LoggerFactory.getLogger(DockerImageBuilder.class)
```

Comment:
built logger does not log to csar's log object.
I propose to get rid of second constructor, and use this signature:
`public DockerImageBuilder(String tag, String dockerWorkDir, TransformationContext context)`
Ofc, then build logger with `context.getLogger()`